src/trace/log.zig

```diff
@@ -1,366 +1,464 @@
-            },
-            .noop => {},
+        pub fn err(self: *const Self) Entry {
+            return switch (self.*) {
+                .noop => Entry{ .noop = {} },
+                inline else => |impl| impl.err(scope),
+            };
         }
-    }
 
-    pub fn info(self: Self, msg: []const u8) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.info(msg);
-            },
-            .noop => {},
+        pub fn warn(self: *const Self) Entry {
+            return switch (self.*) {
+                .noop => Entry{ .noop = {} },
+                inline else => |impl| impl.err(scope),
+            };
         }
-    }
 
-    pub fn debug(self: Self, msg: []const u8) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.debug(msg);
-            },
-            .noop => {},
+        pub fn info(self: *const Self) Entry {
+            return switch (self.*) {
+                .noop => Entry{ .noop = {} },
+                inline else => |impl| impl.info(scope),
+            };
         }
-    }
 
-    pub fn warn(self: Self, msg: []const u8) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.warn(msg);
-            },
-            .noop => {},
+        pub fn debug(self: *const Self) Entry {
+            return switch (self.*) {
+                .noop => Entry{ .noop = {} },
+                inline else => |impl| impl.debug(scope),
+            };
         }
-    }
 
-    pub fn err(self: Self, msg: []const u8) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.err(msg);
-            },
-            .noop => {},
+        pub fn log(self: Self, level: Level, message: []const u8) void {
+            switch (self) {
+                .noop => {},
+                inline else => |*impl| impl.*.log(scope, level, message),
+            }
         }
-    }
 
-    pub fn log(self: Self, level: Level, msg: []const u8) void {
-        switch (self) {
-            inline .standard, .test_logger => |logger| {
-                logger.log(level, msg);
-            },
-            .noop => {},
+        pub fn logf(self: Self, level: Level, comptime fmt: []const u8, args: anytype) void {
+            switch (self) {
+                .noop => {},
+                inline else => |*impl| impl.*.logf(scope, level, fmt, args),
+            }
         }
-    }
+    };
+}
 
-    /// Can be used in tests to minimize the amount of logging during tests.
-    pub const TEST_DEFAULT_LEVEL: Level = .warn;
-};
+pub const Logger = ScopedLogger(null);
 
-pub const StandardErrLogger = struct {
-    allocator: std.mem.Allocator,
-    /// Messages more granular than this will not be logged
+/// An instance of `ScopedLogger` that logs via the channel.
+pub const ChannelPrintLogger = struct {
+    const Self = @This();
```

Comment:
Move `const`(s) below fields as is in the rest of `sig` codebase

Reply:
[2b3fc8a](https://github.com/Syndica/sig/pull/277/commits/2b3fc8a1cc909e2cc11252ba99786104b94c702d)